src/librustc/traits/select.rs

```diff
@@ -2081,6 +2125,11 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 substs.upvar_tys(def_id, self.tcx()).collect()
             }
 
+            ty::TyGenerator(def_id, ref substs, interior) => {
+                let witness = iter::once(interior.witness());
```

Comment:
They can also be type variables and those might be unified with tuples later on. I'm not sure of an easy fix here.